packages/core/src/components/collapse/collapse.tsx

```diff
@@ -117,12 +125,13 @@ export class Collapse extends AbstractComponent<ICollapseProps, ICollapseState>
     }
 
     public render() {
-        const showContents = this.state.animationState !== AnimationStates.CLOSED;
-        const displayWithTransform = showContents && this.state.animationState !== AnimationStates.CLOSING_END;
+        const contentsVisible = this.state.animationState !== AnimationStates.CLOSED;
+        const renderChildren = contentsVisible || this.props.keepChildrenMounted;
```

Comment:
this name suggests a render callback to me: `renderChildren: (args) => JSX.Element`.

`shouldRenderChildren` - obvious boolean.